synapse/lib/httpapi.py

```diff
@@ -326,6 +331,24 @@ async def handleBasicAuth(self):
         self.web_username = udef.get('name')
         return self.web_useriden
 
+    async def handleApiKeyAuth(self):
+        authcell = self.getAuthCell()
+        key = self.request.headers.get('X-API-KEY')
+        isok, info = await authcell.checkUserApiKey(key)  # errfo or dict with tdef + udef
+        if isok is False:
+            self.logAuthIssue(mesg=info.get('mesg'), user=info.get('user'), username=info.get('name'))
+            return
+
+        udef = info.get('udef')
+
+        # Would be nice to know when an API key was being used to auth...
+        # self.logAuthIssue(mesg=f'User access token use: {tdef.get("iden")}',
+        #                   user=udef.get('iden'), username=udef.get('name'), level=logging.DEBUG)
```

Comment:
Is this something to decide on now?

Reply:
I'll remove the note and flag it for revisiting.